logictest: skip `partitioning_hash_sharded_index` in 5node under race

yuzefovich opened the pull request:
We have been seeing this test time out many times under race. I examined a few of the timeouts and didn't find anything suspicious, so let's just skip it under race.

Fixes: #137918.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139297)
<!-- Reviewable:end -->

yuzefovich committed:
logictest: skip `partitioning_hash_sharded_index` in 5node under race

We have been seeing this test time out many times under race. I examined
a few of the timeouts and didn't find anything suspicious, so let's just
skip it under race.

Release note: None

DrewKimball reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139297#-:-OHo88MZCgDT0c11LBNq:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @mgartner)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
